packages/impit-client/src/index.ts

```diff
@@ -20,65 +26,72 @@ export class ImpitHttpClient implements BaseHttpClient {
     private maxRedirects: number;
     private followRedirects: boolean;
 
-    constructor(options?: Omit<ImpitOptions, 'proxyUrl'> & { maxRedirects?: number }) {
-        this.impitOptions = options ?? {};
+    private clientCache: LruCache<Impit> = new LruCache({ maxLength: 10 });
 
-        this.maxRedirects = options?.maxRedirects ?? 10;
-        this.followRedirects = options?.followRedirects ?? true;
-    }
+    private getClient(options: ImpitOptions) {
+        const cacheKey = JSON.stringify(options);
```

Comment:
1. are all the options JSON-serializable?
2. could we use some serialization function that sorts the keys in the options object so that it's more robust?

Reply:
1 - yes

2 - good point! Unfortunately, in modern JS, string keys are enumerated in order of their addition, so it can be quite arbitrary. We are left with something like this:

```suggestion
const cacheKey = JSON.stringify(
     Object.entries(options).sort(([keyA], [keyB]) => {
        if (keyA < keyB) return -1;
        if (keyA > keyB) return 1;
        return 0;
    }).map(([_, x]) => x),
```

wdyt? I don't really like having this here, but we don't use objects like this anywhere else in Crawlee, so extracting this into a function / using a 3rd-party lib seems a bit over the top.